WordPress/Classes/ViewRelated/Feature Highlight/Tooltip.swift

```diff
@@ -0,0 +1,290 @@
+        setUpContainerView()
+        setUpConstraints()
+        isAccessibilityElement = true
+    }
+
+    private func setUpContainerView() {
+        containerView.backgroundColor = .invertedSystem5
+        containerView.layer.cornerRadius = Constants.cornerRadius
+        containerView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(containerView)
+
+        containerTopConstraint = containerView.topAnchor.constraint(equalTo: topAnchor)
+        containerBottomConstraint = bottomAnchor.constraint(equalTo: containerView.bottomAnchor)
+
+        NSLayoutConstraint.activate([
+            containerTopConstraint!,
+            containerView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
+            containerBottomConstraint!
+        ])
+    }
+
+    private func setUpConstraints() {
+        contentStackView.translatesAutoresizingMaskIntoConstraints = false
+        buttonsStackView.translatesAutoresizingMaskIntoConstraints = false
+        containerView.addSubview(contentStackView)
+
+        NSLayoutConstraint.activate([
+            contentStackView.topAnchor.constraint(
+                equalTo: containerView.topAnchor,
+                constant: Constants.Spacing.contentStackViewTop
+            ),
+            contentStackView.leadingAnchor.constraint(
+                equalTo: containerView.leadingAnchor,
+                constant: Constants.Spacing.contentStackViewHorizontal
+            ),
+            containerView.trailingAnchor.constraint(
+                equalTo: contentStackView.trailingAnchor,
+                constant: Constants.Spacing.contentStackViewHorizontal
+            ),
+            containerView.bottomAnchor.constraint(
+                equalTo: contentStackView.bottomAnchor,
+                constant: Constants.Spacing.contentStackViewBottom
+            ),
+            containerView.widthAnchor.constraint(lessThanOrEqualToConstant: Constants.maxWidth),
+            buttonsStackView.heightAnchor.constraint(equalToConstant: Constants.Spacing.buttonStackViewHeight)
+        ])
+    }
+
+    private static func height(
+        withTitle title: String?,
+        message: String?) -> CGFloat {
+            var totalHeight: CGFloat = 0
+
+            totalHeight += Constants.Spacing.contentStackViewTop
+
+            if let title = title {
+                totalHeight += title.height(withMaxWidth: Constants.maxWidth, font: Constants.Font.title)
+            }
+
+            totalHeight += Constants.Spacing.contentStackViewInterItemSpacing
+
+            if let message = message {
+                totalHeight += message.height(withMaxWidth: Constants.maxWidth, font: Constants.Font.message)
+            }
+
+            totalHeight += Constants.Spacing.buttonStackViewHeight
+            totalHeight += Constants.Spacing.contentStackViewBottom
+
+            return totalHeight
+        }
+}
+
+private extension String {
```

Comment:
Is there such an extension we already have? If so I can reuse the existing one. And if not, we might make this public/internal and move it a common file along with its `width` sibling.